Makefile

```diff
@@ -1,35 +1,61 @@
 YQ ?= "yq"
+.phony: test lint autocorrect api_key update_config build 
+.phony: release_dev release build_gem install refresh_config build-docker run-docker exec_app tag
 
-test:
-	bundle exec rspec spec
+config/connectors.yml:
+	cp config/connectors.yml.example config/connectors.yml
 
-lint:
-	bundle exec rubocop lib spec
+test: config/connectors.yml
+	bundle _$(shell cat .bundler-version)_ exec rspec spec
 
-autocorrect:
-	bundle exec rubocop lib spec -a
+lint: config/connectors.yml
+	bundle _$(shell cat .bundler-version)_ exec rubocop lib spec
 
-api_key:
+autocorrect: config/connectors.yml
+	bundle _$(shell cat .bundler-version)_ exec rubocop lib spec -a
+
+api_key: config/connectors.yml
 	${YQ} e ".http.api_key = \"$(shell uuidgen | tr -d '-')\"" -i config/connectors.yml
 
 # build will set the revision key in the config we use in the Gem
 # we can add more build=time info there if we want
-build:
+update_config_dev: config/connectors.yml
+	${YQ} e ".revision = \"$(shell git rev-parse HEAD)\"" -i config/connectors.yml
+	${YQ} e ".repository = \"$(shell git config --get remote.origin.url)\"" -i config/connectors.yml
+	${YQ} e ".version = \"$(shell script/version.sh)\"" -i config/connectors.yml
+
+update_config: config/connectors.yml
+	${YQ} e ".revision = \"$(shell git rev-parse HEAD)\"" -i config/connectors.yml
+	${YQ} e ".repository = \"$(shell git config --get remote.origin.url)\"" -i config/connectors.yml
+	${YQ} e ".version = \"$(shell cat VERSION)\"" -i config/connectors.yml
```

Comment:
In `ent-search` repo, we have [enterprise-search.yml](https://github.com/elastic/ent-search/blob/main/script/togo/enterprise-search.yml), and you can copy a local yml file (enterprise-search.local.yml) from here. Should we use the same pattern in connectors?

Reply:
not sure what you are proposing

Reply:
Can we have `config/connectors.yml` instead of `config/connectors.yml.example`?

You can copy a `config/connectors.local.yml` for local development. We can default to `config/connectors.yml` for all configs in Makefile, and use `config/connectors.local.yml` for `release_dev`?

Reply:
I'm not sure it's something that is needed for today - it's different from how it works in Enterprise Search. `enterprise-search.yml` config file is actually a redistributable config file that is used by docker or actual distribution, and it's possible to actually run Enterprise Search with this file.

For connectors the file cannot be used out-of-the box, cause it needs additional setup - such as api key, for example.

In future when we decide for example to run connectors with docker, we can move closer to this approach (having `config/connectors.yml` instead of `config/connectors.yml.example`, for now I think it's okay to leave it the way it is.